spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/impl/DefaultSchedulerService.java

```diff
@@ -137,14 +146,52 @@ public void unschedule(String scheduleName) {
 
 	@Override
 	public List<ScheduleInfo> list(Pageable pageable, String taskDefinitionName) {
-		// Need to add support for pagination
-		return scheduler.list(taskDefinitionName);
+		throw new UnsupportedOperationException("method not supported");
 	}
 
 	@Override
 	public List<ScheduleInfo> list(Pageable pageable) {
-		// Need to add support for pagination
-		return scheduler.list();
+		throw new UnsupportedOperationException("method not supported");
+	}
+
+	@Override
+	public List<ScheduleInfo> list(String taskDefinitionName) {
+		return limitScheduleInfoResultSize(scheduler.list(taskDefinitionName),
+				this.schedulerServiceProperties.getMaxSchedulesReturned());
+	}
+
+	@Override
+	public List<ScheduleInfo> list() {
+		return limitScheduleInfoResultSize(scheduler.list(),
+				this.schedulerServiceProperties.getMaxSchedulesReturned());
+	}
+
+	private List<ScheduleInfo> limitScheduleInfoResultSize(List<ScheduleInfo> resultSet,
+			int schedulerLimitResultSize) {
+		if(resultSet.size() > schedulerLimitResultSize) {
+			resultSet = resultSet.subList(0, schedulerLimitResultSize);
+		}
+		return resultSet;
+	}
+
+	/**
+	 * Retain only properties that are meant for the <em>scheduler</em> of a given task(those
+	 * that start with {@code scheduler.}and qualify all
+	 * property values with the {@code spring.cloud.scheduler.} prefix.
+	 *
+	 * @param input the scheduler properties
+	 * @return scheduler properties for the task
+	 */
+	private static Map<String, String> extractAndQualifySchedulerProperties(Map<String, String> input) {
+		final String prefix = "scheduler.";
```

Comment:
This is the prefix that needs to be validated during the validation not the `spring.cloud.scheduler` one.